libnavui-lifecycle/src/main/java/com/mapbox/navigation/ui/lifecycle/MapboxNavigationObserver.kt

```diff
@@ -0,0 +1,65 @@
+package com.mapbox.navigation.ui.lifecycle
+
+import com.mapbox.navigation.base.ExperimentalPreviewMapboxNavigationAPI
+import com.mapbox.navigation.core.MapboxNavigation
+
+/**
+ * This allows you to attach and detach [MapboxNavigation] observers
+ * from any component with a lifecycle.
+ *
+ * Example of observing locations with a view model
+ * ```
+ * class MyViewModel : ViewModel() {
+ *   private val locationObserver = LocationObserver()
+ *
+ *   val location: LiveData<Location> = locationObserver.location.asLiveData()
```

Comment:
```suggestion
 *   private val locationObserver = LocationObserver()
 *
 *   val location: LiveData<Location> = locationObserver.location.asLiveData()
 *
 * val myMapboxNavigationObserver = MyMapboxNavigationObserver()
```

Reply:
This might also need a cleanup of how location actually gets to `val location: LiveData<Location>`

Reply:
Yeah we may also build some tools on top of the callbacks to make it simpler. Goal is to make it so we can [simplify customer code like this](https://github.com/mapbox/mapbox-navigation-android-examples/blob/2878edfb3fcbac0a8db8266f0d5f156267d2ccf5/app/src/main/java/com/mapbox/navigation/examples/basics/ShowCurrentLocationActivity.kt#L51-L86) 

Here's the version we're using in android auto [CarAppLocationImpl](https://github.com/mapbox/mapbox-navigation-android-examples/blob/main/android-auto/src/main/java/com/mapbox/androidauto/navigation/location/impl/CarAppLocationImpl.kt)